### ISSUE TITLE
version upgrade for ckan2.9

### ISSUE THREAD

**Issue by liquanyong305** [conwetlab/ckanext-oauth2#45]
Hello everyone

We use Ckan2.9.4 to develop our project, but ckan-oauth2 only supports 2.7 or 2.8.
Do you have any plans to upgrade version for ckan2.9?

Thanks a lot for the work you have done on this plugin.

**Comment by frafra** [comment]
Please test and give feedback to #46.

**Comment by Nisha1293** [comment]
Hi @frafra,

I tested #46 on latest CKAN v2.9.9 and v2.10.1 but I am getting below error:
AttributeError
AttributeError: 'Request' object has no attribute 'GET'

Any suggestion on this? Does any other PR exist, or are the latest changes in another branch?"

**Comment by frafra** [comment]
@nisha-nec See #42.

**Comment by Nisha1293** [comment]
Hi @frafra 
Thanks for the information.
I tried but the code changes as per the CKANv2.9.x. 
I tested it on CKANv2.10.1, in this version,  Authentication logic to use [Flask-login] instead of repoze.who.
Do you have updated code. ?
Any help will be appreciated.

**Comment by frafra** [comment]
@nisha-nec I use this one with CKAN 2.9: https://github.com/NINAnor/ckanext-oauth2/tree/aeae71acd20dffd49a3eb7b084a32523859b3212

**Comment by Nisha1293** [comment]
@frafra  yes, in 2.9 it will work fine, but from CKAN 2.10, Repoze has been replaced for flask-login and this includes several changes:

1. Login is now handled by user_login function (instead of set_repoze_user)
2. auth_tkt cookie has been deprecated since it was being set by Repoze

**Comment by frafra** [comment]
@nisha-nec Sorry for the misunderstanding. As you can see, no one seems interested in supporting OAuth2 on newer CKAN. Someone needs to pick up the PR for 2.9, fix the remaining issues, then merge the PR (or fork the repository), and then focus on the 2.10.
